tasks.py

```diff
@@ -79,6 +79,63 @@ def create_web_resource(resource_name, src_dir):
     )
 
 
+@invoke.task(help={
+    'python-version': (
+        'Can be used to define the python version used when creating the'
+        ' work environment'
+    ),
+})
+def docs(ctx, python_version=None):
+    """
+    Create the documentation html locally.
+    """
+    import json
+    import subprocess
+    import tempfile
+    from pathlib import Path
+
+    conda_info_json = subprocess.check_output(['conda', 'info', '--json'])
+    conda_info = json.loads(conda_info_json)
+    current_env_name = conda_info["active_prefix_name"]
+    if current_env_name in (None, 'base'):
+        raise invoke.Exit("Activate the project's conda environment first")
+    else:
+        docs_env_name = f'{current_env_name}-docs'
+
+    new_environ = os.environ.copy()
+    new_environ['TEST_QMXGRAPH'] = '0'
+    if python_version is not None:
+        new_environ['PYTHON_VERSION'] = python_version
+
+    script = [
+        '',  # To have a new line at the start (see windows new line).
+        f'conda devenv --name {docs_env_name} --file docs_environment.devenv.yml',
+        f'conda activate {docs_env_name}',
+        'cd docs',
+        'sphinx-build . _build -W',
+    ]
+    if sys.platform == 'win32':
+        suffix = '.bat'
+        new_line = '\n@echo on\ncall '
+        command = ['cmd', '/C']
+    else:
+        suffix = '.bash'
+        new_line = '\n'
+        command = ['bash', '-x']
+
+    script_file = tempfile.NamedTemporaryFile(suffix=suffix, delete=False)
```

Comment:
Why are you passing `delete=False` and then using `os.unlink()`? Unless there's a reason it would be better to use this as a context manager, with `delete=True`.

Reply:
Because I need to be able to close the file to execute the `script_file ` if the file is removed when it is closed.

Reply:
With 
```python
    with tempfile.NamedTemporaryFile(mode='w', suffix=suffix) as script_file:
        script_file.write(new_line.join(script))
        script_file.flush()

        command.append(script_file.name)
        subprocess.check_call(command, env=new_environ)
```
```shell
(qmxgraph) λ inv docs
The process cannot access the file because it is being used by another process.
```

With
```python
    with tempfile.NamedTemporaryFile(mode='w', suffix=suffix) as script_file:
        script_file.write(new_line.join(script))
        script_file.close()

        command.append(script_file.name)
        subprocess.check_call(command, env=new_environ)
```
```shell
(qmxgraph) λ inv docs
'W:\my\tmp\tmpvv57sfdx.bat' is not recognized as an internal or external command,
operable program or batch file.
```

Reply:
Ahh of course